### ISSUE TITLE
sane generate resource [singular | plural] debate

### ISSUE THREAD

**Issue by jamemackson** [artificialio/sane#123]
It seems there is an active debate going on if we should be generating resources with sane using the singular or plural version of the noun in question.  

Here are the versions of each component here I'm using
``` Bash
jame-dev:borrowers jame$ sane --version
0.0.24
jame-dev:borrowers jame$ sails --version
0.11.0-rc10
jame-dev:borrowers jame$ ember --version
version: 0.2.1
node: 0.12.0
npm: 2.7.3
```

As of this writing when 
`sane generate resource friends firstName:string lastName:string etc` 
is being called this is the output:

``` Bash
jame-dev:borrowers jame$ sane generate resource friends firstName:string lastName:string email:string twitter:string totalArticles:number
info: Created a new model ("Friends") at api/models/Friends.js!
info: Created a new controller ("friends") at api/controllers/FriendsController.js!
version: 0.2.1
installing
  create app/models/friend.js
installing
  create tests/unit/models/friend-test.js
installing
  create app/routes/friends.js
  create app/templates/friends.hbs
installing
  create tests/unit/routes/friends-test.js
```

so this is creating the sails model and controller with plural, and the ember route and template as plural but the ember model is singular.

now making the same call using the singular 'friend' results in this:
``` bash
jame-dev:borrowers jame$ sane generate resource friend firstName:string lastName:string email:string twitter:string totalArticles:number
info: Created a new model ("Friend") at api/models/Friend.js!
info: Created a new controller ("friend") at api/controllers/FriendController.js!
version: 0.2.1
0.2.1

installing
  create app/models/friend.js
installing
  create tests/unit/models/friend-test.js
installing
  create app/routes/friend.js
  create app/templates/friend.hbs
installing
  create tests/unit/routes/friend-test.js
```

so this created everything in the singular... which I believe (and correct me if I'm wrong here) is an issue with how the ember rest adapter expects the api endpoints to be plural.  see: http://emberjs.com/guides/models/the-rest-adapter/ 

So it seems to me that we would want the sails model and controller to be plural, and the Ember model to be singular by their convention.  I'm unsure what the standard should be for the route and template on the ember side but, for myself at least, in this case I would want the user to go to `/friends` to see the list of friends and then to drill into a specific friend by going to `/friend/:friend_id` Although going to `/friends/:friend_id` is not a terrible alternative either.  I'd rather that than having the user go to `/friend` to see the list of friends.

So I'm starting this issue thread in hopes that we can identify the right mix here and settle this debate. :)

Cheers all!

**Comment by mgenev** [comment]
Thanks for the write up. In addition to all this, I believe there's also the behavior that the Sails blueprint we use will pluralize singular endpoints itself.

**Comment by seriousben** [comment]
I think it should follow http://jsonapi.org/ which Ember-data will always support.

**Comment by mgenev** [comment]
Ember data doesn't support json api yet as far as I know...

**Comment by Globegitter** [comment]
Yep, Ember data does not fully support json api as of yet. And the way I see it in terms of singular/plural. Create your models in singular, you are representing one instance of you model and let the tools care about the rest. The pluralization etc.

And with your example if you `sane g api friend` it would be available under `/friends` and `/friends/:friend_id`. Reason for that being that you don't have two different enpoints for the same resource which in my opinion would be more confusing.

**Comment by mgenev** [comment]
The thing is according to what @jamemackson posted, if you do the singular generation with sane, you end up with everything singular in Ember as well, which to me is odd, I'd want my browser route to say /vendors  to get the index, not /vendor. This is actually what I had brought up a few months ago but forgot about since I had started to generate stuff separately because of it. Thoughts? Am I missing something?

**Comment by Globegitter** [comment]
Yeah one thing we can look into is when you create `sane g api friends` that we pass on the singular word to Sails. I haven't looked into how sails behaves then. If the route then is `/friends/` or stupidly `/friendss` since it just tries to pluralize the `Friends.js` model.

**Comment by IanVS** [comment]
So, it seems that the solution is to use the plural when you generate the resource, except that the model in Sails is generated plural.  Ember behaves as we would expect and singularizes the model.  Sails is the issue, no?

**Comment by mgenev** [comment]
It sounds to me that ideally sane-cli should do sails singular and ember plural

**Comment by IanVS** [comment]
@mgenev but then the Sails controller would be singular as well.  Does that matter?

**Comment by mgenev** [comment]
I think not, because the ember blueprint  pluralizes

**Comment by Globegitter** [comment]
Just got over that in my app and https://github.com/artificialio/sane/issues/123#issuecomment-86626471 is indeed the best thing to do. @jamemackson feel up for a PR for that? Should be fairly straight-forward.

**Comment by jamemackson** [comment]
@Globegitter I'm happy to give it a shot.  This would be to change sane so it pluralizes the sails side correct?  I may hit you up for some guidance before I set out on this one though.

**Comment by Globegitter** [comment]
@jamemackson yeah of course. The file to change will be https://github.com/artificialio/sane/blob/master/lib/commands/generate.js and you probably want to look into https://github.com/blakeembrey/pluralize or https://github.com/stefanpenner/ember-inflector. And of course just hit me up on gitter - thanks :)
